src/components/dashboard/DashboardPage.jsx

```diff
@@ -38,6 +39,17 @@ const DashboardPage = () => {
     },
   } = useEnterpriseCuration(enterpriseConfig.uuid);
 
+  const hasActiveCourseAssignments = (learnerCreditPolicies) => {
+    const learnerContentAssignmentsArray = learnerCreditPolicies?.flatMap(
+      item => item?.learnerContentAssignments || [],
+    );
```

Comment:
[suggestion, non-blocking] Looks like logic to generate a flat map of assignments based on the learner credit policies happens in two places now. I'm wondering if it'd be worth abstracting this logic out to where we get the response about redeemable learner credit policies from the `credits_available` API so components such as this one don't need to run their own `.flatMap` to get a list of assignments.

E.g., `useRedeemableLearnerCreditPolicies` and its query function `getRedeemablePoliciesData` could return a data structure more along the lines of the following, where it splits out `redeemablePolicies` vs. `learnerContentAssignments`:

```jsx
const getRedeemablePoliciesData = async ({ queryKey }) => {
  const enterpriseId = queryKey[1];
  const userID = queryKey[2];
  const response = await fetchRedeemableLearnerCreditPolicies(enterpriseId, userID);
  const redeemablePolicies = camelCaseObject(transformRedeemablePoliciesData(response.data));
  const learnerContentAssignments = redeemablePolicies.flatMap(policy => policy.learnerContentAssignments);
  return {
    redeemablePolicies,
    learnerContentAssignments,
  };
};

export function useRedeemableLearnerCreditPolicies(enterpriseId, userID) {
  return useQuery({
    queryKey: ['redeemablePolicies', enterpriseId, userID],
    queryFn: getRedeemablePoliciesData,
    onError: (error) => {
      logError(error);
    },
  });
}
```

Any component that relies on an assignments list could rely on `learnerContentAssignments` returned by the hook instead of having to re-compute the `.flatMap` each time. Then, the same line in the `isDisableCourseSearch` function could be simplified to:

```jsx
const assignments = redeemableLearnerCreditPolicies?.learnerContentAssignments;
```

Reply:
Opted to resolve this in a followup PR to focus on the MVP, but intended to resolve immediately following the validation of functionality of this PR in stage/prod environment. Updated ticket to document the decision.